service/javadsl/kafka/server/src/main/scala/com/lightbend/lagom/internal/javadsl/broker/kafka/JavadslRegisterTopicProducers.scala

```diff
@@ -98,11 +98,10 @@ class JavadslRegisterTopicProducers @Inject() (
                   projectionRegistryImpl
                 )
 
-              case other =>
-                log.warn {
-                  s"Unknown topic producer ${other.getClass.getName}. " +
-                    s"This will likely result in no events published to topic ${topicId.value} by service ${info.serviceName}."
-                }
+              case null =>
+                log.error (
+                  s"Expected Topic type ${classOf[MethodTopicHolder]}, but null instance was passed" ,new NullPointerException()
```

Comment:
Thanks @Munandermaan. 

Sorry, I should have been more precise. The message is saying it expected a Topic type, but it will be better to say: 

`s"Expected an instance of ${classOf[MethodTopicHolder]}, but 'null' was passed"`

A few more things...

Maybe we should add a message to the `new NullPointerException()` constructor. That will help users that log that exception elsewhere. You can re-use the logging message and pass it to the constructor.

Could you also have a look on that Scala DSL variation? We may also need to add that change there.

Reply:
Done